src/prefect/orion/database/migrations/versions/postgresql/2023_01_13_045500_2882cd2df463_implement_artifact_table.py

```diff
@@ -0,0 +1,180 @@
+"""Implements the artifact table and migrates run results
+
+Revision ID: 2882cd2df463
+Revises: d481d5058a19
+Create Date: 2023-01-13 04:55:00.358638
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+import prefect
+
+# revision identifiers, used by Alembic.
+revision = "2882cd2df463"
+down_revision = "d481d5058a19"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.create_table(
+        "artifact",
+        sa.Column(
+            "id",
+            prefect.orion.utilities.database.UUID(),
+            server_default=sa.text("(GEN_RANDOM_UUID())"),
+            nullable=False,
+        ),
+        sa.Column(
+            "created",
+            prefect.orion.utilities.database.Timestamp(timezone=True),
+            server_default=sa.text("CURRENT_TIMESTAMP"),
+            nullable=False,
+        ),
+        sa.Column(
+            "updated",
+            prefect.orion.utilities.database.Timestamp(timezone=True),
+            server_default=sa.text("CURRENT_TIMESTAMP"),
+            nullable=False,
+        ),
+        sa.Column("key", sa.String(), nullable=True),
+        sa.Column("artifact_type", sa.String(), nullable=True),
+        sa.Column("artifact_data", sa.JSON(), nullable=True),
+        sa.Column("artifact_metadata", sa.JSON(), nullable=True),
```

Comment:
Any particular reason you chose to prefix these columns with the table name?

Reply:
I don't think any serious ones, I think of these names were reserved by pydantic or sqlalchemy or something so I just added a prefix. I think there is a way to get around that pretty easily if we want it

Reply:
Renamed cols `artifact_type` to `type`, `artifact_data` to `data`, `artifact_metadata` to `metadata`.

Reply:
Changed col name to `metadata_`.